ui/web-v2/apps/admin/src/components/FeatureTargetingForm/index.tsx

```diff
@@ -312,9 +312,7 @@ export const PrerequisiteInput: FC<PrerequisiteInputProps> = memo(
     };
 
     useEffect(() => {
-      if (prerequisites?.length > 0) {
-        dispatchListFeatures();
-      }
+      dispatchListFeatures();
```

Comment:
@bimalgrg519 

This way the dispatch will always trigger when the page is accessed.

We want to dispatch it when the page is loaded and it has prerequisites.
Or when it doesn't have prerequisites, we want to dispatch it when the `Add Prerequisites` button is clicked.

Reply:
@cre8ivejp  Yeah i agree with your points. But to make add Prerequisites button enable or disable at first, we need total numbers of feature flag. That's why we need to dispatch feature flag list api.

Reply:
@bimalgrg519 
Can't you can solve this by adding a condition inside of `disableAddPrerequisite`?

```ts
useEffect(() => {
  if (prerequisites.length > 0) {
    dispatchListFeatures();
  }
}, []);

const disableAddPrerequisite = useCallback(() => {
  if (prerequisites.length === 0) {
    return false;
  }
  return prerequisites.length === features.length - 1;
}, [prerequisites, features]);
```

Reply:
Thank you so much @cre8ivejp . Perfect solution.